docat/docat/utils.py

```diff
@@ -109,22 +109,24 @@ def get_all_projects(upload_folder_path: Path, include_hidden: bool) -> Projects
     Returns all projects in the upload folder.
     """
     projects: list[Project] = []
-
     for project in upload_folder_path.iterdir():
         if not project.is_dir():
             continue
 
-        details = get_project_details(upload_folder_path, project.name, include_hidden)
-
-        if details is None:
-            continue
+        if not include_hidden:
+            # we need to ignore hidden versions, because if the only version is hidden, the project should be hidden altogether
+            tags_and_versions = [x for x in project.iterdir() if x.is_dir() and not (x / ".hidden").exists()]
+        else:
+            tags_and_versions = [x for x in project.iterdir() if x.is_dir()]
 
-        if len(details.versions) < 1:
+        version_count = len(tags_and_versions)
```

Comment:
Why exactly do we only return the count when we load the versions anyway?

Reply:
It's just to show a number while they aren't loaded yet. We could remove it theoretically, but we'd have to see that the UI updates when the versions are finally loaded.